From acfc2a44e196c03f41da0276ced1e9c4da67362e Mon Sep 17 00:00:00 2001
From: Stefano Babic <sbabic@denx.de>
Date: Fri, 30 Mar 2012 10:25:18 +0200
Subject: [PATCH 2/4] OMAP3: mcx: set pinmux for uart4

Signed-off-by: Stefano Babic <sbabic@denx.de>
CC: Tom Rini <trini@ti.com>
---
 board/htkw/mcx/mcx.h |   11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/board/htkw/mcx/mcx.h b/board/htkw/mcx/mcx.h
index d675a48..867cc9e 100644
--- a/board/htkw/mcx/mcx.h
+++ b/board/htkw/mcx/mcx.h
@@ -284,9 +284,14 @@ const omap3_sysinfo sysinfo = {
 					/* HSUSB2_dat3 */\
 	/* CCDC */\
 	MUX_VAL(CP(CCDC_PCLK),		(IEN  | PTD | EN  | M4)) \
-	MUX_VAL(CP(CCDC_FIELD),		(IEN  | PTD | EN  | M4)) \
-	MUX_VAL(CP(CCDC_HD),		(IEN  | PTD | EN  | M4)) \
-	MUX_VAL(CP(CCDC_VD),		(IEN  | PTD | EN  | M4)) \
+	/* CCDC_FIELD: gpio_95, uP-TXD4 */ \
+	MUX_VAL(CP(CCDC_FIELD),		(IDIS | PTD | DIS | M2)) \
+	/* CCDC_HD: gpio_96, uP-RTS4# */ \
+	MUX_VAL(CP(CCDC_HD),		(IDIS | PTD | DIS | M2)) \
+	/* CCDC_VD: gpio_97, uP-CTS4# */ \
+	MUX_VAL(CP(CCDC_VD),		(IEN  | PTD | EN  | M2)) \
+	/* CCDC_WEN: gpio_98, uP-RXD4 */ \
+	MUX_VAL(CP(CCDC_WEN),		(IEN  | PTD | DIS | M2)) \
 	MUX_VAL(CP(CCDC_WEN),		(IEN  | PTD | EN  | M4)) \
 	MUX_VAL(CP(CCDC_DATA0),		(IEN  | PTD | EN  | M4)) \
 	MUX_VAL(CP(CCDC_DATA1),		(IEN  | PTD | EN  | M4)) \
-- 
1.7.9.5

